lib/class-wp-duotone-gutenberg.php

```diff
@@ -939,9 +939,17 @@ public static function output_footer_assets() {
 			echo self::get_svg_definitions( self::$used_svg_filter_data );
 		}
 
-		// This is for classic themes - in block themes, the CSS is added in the head via wp_add_inline_style in the wp_enqueue_scripts action.
-		if ( ! wp_is_block_theme() && ! empty( self::$used_global_styles_presets ) ) {
-			wp_add_inline_style( 'core-block-supports', self::get_global_styles_presets( self::$used_global_styles_presets ) );
+		// In block themes, the CSS is added in the head via wp_add_inline_style in the wp_enqueue_scripts action.
+		if ( ! wp_is_block_theme() ) {
+			$style_tag_id = 'core-block-supports-duotone';
+			wp_register_style( $style_tag_id, false );
+			if ( ! empty( self::$used_global_styles_presets ) ) {
+				wp_add_inline_style( $style_tag_id, self::get_global_styles_presets( self::$used_global_styles_presets ) );
+			}
+			if ( ! empty( self::$block_css_declarations ) ) {
+				wp_add_inline_style( $style_tag_id, wp_style_engine_get_stylesheet_from_css_rules( self::$block_css_declarations ) );
```

Comment:
Should the prefix of `wp_style_engine_get_stylesheet_from_css_rules()` function be changed to `gutenberg_`?

Reply:
> Should the prefix of `wp_style_engine_get_stylesheet_from_css_rules()` function be changed to `gutenberg_`?

Yeah, the `gutenberg_` prefixed functions should be preferred over the `wp_` ones if they exist in the plugin. This way we're testing the changes to those functions everywhere within the plugin. That way issues are much more likely to surface before porting things to core.